src/hooks/useFormField.tsx

```
@@ -0,0 +1,70 @@
+import { useFormContext, useWatch } from "react-hook-form";
+import { getValidationRules } from "@/utils/getValidationRules";
+import { useEffect } from "react";
+
+interface UseFormFieldProps {
+  name: string;
+  type: string;
+  label?: string;
+  required?: boolean;
+  defaultValue?: any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)